### PR TITLE
CC-1212 Improve build status fetching + added sentry capture

### PR DESCRIPTION
**Pull request by libmartinito**

### CODE REVIEW

File: internal/utils/codecrafters_client.go [hunk 1]
```diff
@@ -210,26 +222,34 @@ func (c CodecraftersClient) FetchBuild(buildId string) (FetchBuildStatusResponse
...
-
-	fetchBuildResponse := FetchBuildStatusResponse{}
+	attempts := 0
+	buildStatus := "not_started"
+	var fetchBuildResponse FetchBuildStatusResponse
 
-	err = json.Unmarshal(response.Bytes(), &fetchBuildResponse)
-	if err != nil {
-		return FetchBuildStatusResponse{}, fmt.Errorf("failed to fetch build result from CodeCrafters: %s", err)
+	for buildStatus != "success" && buildStatus != "failure" && buildStatus != "error" && attempts < 20 {
```
Comment (rohitpaulk) [inline internal/utils/codecrafters_client.go:229]: @libmartinito are we not retrying this already?

File: internal/utils/codecrafters_client.go [hunk 2]
```diff
@@ -190,6 +199,9 @@ func (c CodecraftersClient) FetchBuild(buildId string) (FetchBuildStatusResponse
 			}
 
 			if fetchBuildResponse.Status != "failure" && fetchBuildResponse.Status != "success" {
+				customError := &CustomError{Message: "unexpected submission status: " + fetchBuildResponse.Status}
+				sentry.CaptureException(customError)
```
Comment (rohitpaulk) [inline internal/utils/codecrafters_client.go:203]: Why is `CustomError` important here? If it's just carrying `Message`, can't we use `fmt.Errorf`?

File: internal/utils/codecrafters_client.go [hunk 3]
```diff
@@ -190,6 +199,9 @@ func (c CodecraftersClient) FetchBuild(buildId string) (FetchBuildStatusResponse
...
+
```
Comment (rohitpaulk) [inline internal/utils/codecrafters_client.go:204]: We probably need monitoring here too btw: https://github.com/codecrafters-io/test-runner/blob/44ea70a6e03cc72f4a346a7f38636f4ccb2c2472/handle_git_push_command.go#L147

**Comment by libmartinito** [comment]
remade pr [here](https://github.com/codecrafters-io/cli/pull/24)

**Pull request closed without merging**